src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/LibraryRepository.kt

```diff
@@ -1,6 +1,83 @@
 package systems.ajax.englishstudytelegrambot.repository
 
-import org.springframework.data.mongodb.repository.MongoRepository
+import org.bson.types.ObjectId
+import org.springframework.dao.DuplicateKeyException
+import org.slf4j.LoggerFactory
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.stereotype.Repository
 import systems.ajax.englishstudytelegrambot.entity.Library
+import systems.ajax.englishstudytelegrambot.entity.Word
+import systems.ajax.englishstudytelegrambot.exception.LibraryWithTheSameNameForUserAlreadyExistException
+import systems.ajax.englishstudytelegrambot.exception.LibraryIsMissingException
 
-interface LibraryRepository : MongoRepository<Library, String>
+interface LibraryRepository {
+
+    fun saveNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Library
+
+    fun getAllLibraries(): List<Library>
+
+    fun deleteLibrary(nameOfLibraryForDeleting: String, telegramUserId: String): Library
```

Comment:
I don't see any constraints that nameOfLibrary is unique, so it's probably better to delete the library by its id rather than name

Reply:
pair of libraryName and telegramUserId in library collection is always unique. I made even index for that. I can rewrite to making two operation with db: 
1) find libraryId by pair (library name and telegramUserId)
2) delete library by libraryId
or would it better to perform it by using only one operation - find by two criterias and delete ?
this problem  looks like : readable vs performance